### PR TITLE
CIV-12792 go to hearing

### PR DESCRIPTION
**Pull request by paul-pearson**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-12792

### Change description ###

go to hearing

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/dashboard/scenarios/claimant/GoToHearingScenarioTest.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        //Verify Notification is created
+        doGet(BEARER_TOKEN, GET_NOTIFICATIONS_URL, caseId, "CLAIMANT")
+            .andExpect(status().isOk())
+            .andExpectAll(
+                status().is(HttpStatus.OK.value()),
+                jsonPath("$[0].titleEn").value("Wait for the court to review the case"),
+                jsonPath("$[0].descriptionEn")
+                    .value("You have rejected ${defendantName}'s response and want to proceed to court." +
+                               " If the case goes to a hearing we will contact you with further details. " +
+                               "<a href=\"{VIEW_RESPONSE_TO_CLAIM}\"  " +
```
Comment (dharmendrak) [inline src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/dashboard/scenarios/claimant/GoToHearingScenarioTest.java:41]: there is line gap before responseToclaim link

File: src/main/resources/db/migration/V2024_02_02_12356_1__create_dashboard_notification_template_table.sql [hunk 2]
```diff
@@ -7,8 +7,8 @@ CREATE TABLE IF NOT EXISTS dbs.dashboard_notifications_templates (
  template_name VARCHAR(256),
  title_en VARCHAR(256),
  title_cy VARCHAR(256),
- description_en VARCHAR(256),
- description_cy VARCHAR(256),
+ description_en VARCHAR(512),
+ description_cy VARCHAR(512),
```
Comment (dharmendrak) [inline src/main/resources/db/migration/V2024_02_02_12356_1__create_dashboard_notification_template_table.sql:11]: we are going to make it 1024

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/claimant/GoToHearingNotificationHandler.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    private CallbackResponse configureScenarioForGoToHearing(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        if (callbackParams.getCaseData().getCcdState() == CaseState.JUDICIAL_REFERRAL) {
```
Comment (kannan-v-hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/claimant/GoToHearingNotificationHandler.java:52]: CaseData is already present. No need for another getCaseData() call

Comment (paul-pearson) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/claimant/GoToHearingNotificationHandler.java]: fixed

**Review by dharmendrak — APPROVED** [review]

File: src/main/resources/db/migration/V2024_02_02_12356_1__create_dashboard_notification_template_table.sql (hunk 2 above)
Comment (paul-pearson) [reply src/main/resources/db/migration/V2024_02_02_12356_1__create_dashboard_notification_template_table.sql]: changed to 1024

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/claimant/ClaimantResponseNotificationHandler.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.civil.callback.CallbackParams.Params.BEARER_TOKEN;
+import static uk.gov.hmcts.reform.civil.callback.CallbackType.ABOUT_TO_SUBMIT;
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.CREATE_CLAIMANT_DASHBOARD_NOTIFICATION_FOR_CLAIMANT_RESPONSE;
+import static uk.gov.hmcts.reform.civil.handler.callback.camunda.dashboardnotifications.DashboardScenarios.SCENARIO_AAA7_CLAIMANT_INTENT_GO_TO_HEARING;
+
+@Service
+@RequiredArgsConstructor
+public class ClaimantResponseNotificationHandler extends CallbackHandler {
```
Comment (portizdev) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/claimant/ClaimantResponseNotificationHandler.java:27]: Pls use more specific name. ClaimantResponseClaimantNotificationHandler, to be able to differentiate it from defendant handler.

Comment (kannan-v-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/claimant/ClaimantResponseNotificationHandler.java:27]: Hi Pablo. We have different folders for Claimant and Defendant Handlers. Do we still need to specify it again in the file name ?

Comment (portizdev) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/claimant/ClaimantResponseNotificationHandler.java:27]: Sorry, I didn't notice the folder. I think it is still useful to have different names especially to be able to search more easily the files, but I understand that it is already differentiated with the folder. 

I resolve the comment and leave it to your free choice.

**Review by portizdev — APPROVED** [review]

**Pull request merged**